### ISSUE TITLE
Looks like supports is ignored from tar files

### ISSUE THREAD

**Issue by chris-rock** [inspec/inspec#360]
I run the example profile on Mac, where the `inspec.yml` contains

``` yaml
supports:
  - os-family: linux
```

The tests are not executed:

```
$ inspec exec profile.tar.gz
..

Finished in 0.00308 seconds (files took 0.49697 seconds to load)
2 examples, 0 failures
```

If I generate an archive with `inspec archive`, the tests are executed:

```
$ inspec exec examples/profile 


Finished in 0.00042 seconds (files took 0.3386 seconds to load)
0 examples, 0 failures
```

This also highlights the importance for #354. Otherwise the user is not able to track down, why tests are not executed.

**Comment by conorsch** [comment]
Running into this issue with a Linux host as well. Yes, the `inspec.yml` file declares support for Linux. I'm also seeing debug output about `No controls or tests were defined`, even though they're clearly there.

```
W, [2016-02-01T18:15:12.313742 #1573]  WARN -- : Missing profile copyright in inspec.yml
W, [2016-02-01T18:15:12.313824 #1573]  WARN -- : No controls or tests were defined.
I, [2016-02-01T18:15:12.316651 #1573]  INFO -- : Profile check finished. Generate archive /home/conor/.gem/ruby/2.1.0/gems/inspec-0.9.11/elk.tar.gz.
D, [2016-02-01T18:15:12.317509 #1573] DEBUG -- : Add the following files to archive:
D, [2016-02-01T18:15:12.317551 #1573] DEBUG -- :     controls
D, [2016-02-01T18:15:12.317594 #1573] DEBUG -- :     controls/kibana_spec.rb
D, [2016-02-01T18:15:12.317622 #1573] DEBUG -- :     controls/tls_spec.rb
D, [2016-02-01T18:15:12.317650 #1573] DEBUG -- :     controls/logstash_spec.rb
D, [2016-02-01T18:15:12.317679 #1573] DEBUG -- :     controls/nginx_spec.rb
D, [2016-02-01T18:15:12.317706 #1573] DEBUG -- :     controls/elasticsearch_spec.rb
D, [2016-02-01T18:15:12.317733 #1573] DEBUG -- :     inspec.yml
I, [2016-02-01T18:15:12.338399 #1573]  INFO -- : Finished archive generation.
```

Unfortunately, even running against the archive doesn't solve the problem—at least not when I'm using SSH:

```
$ inspec exec --backend ssh --host <REDACTED> --user conor -i ~<REDACTED> /home/conor/.gem/ruby/2.1.0/gems/inspec-0.9.11/elk.tar.gz

[..snip..]
Finished in 0.00027 seconds (files took 8.12 seconds to load)
0 examples, 0 failures

zlib(finalizer): the stream was freed prematurely.
```

For both the archive and the local directory, running them with the local backend shows test discovery working fine. It's only the SSH backend that skips the discovery phase silently.

Let me know if I can provide more data.

**Comment by chris-rock** [comment]
fixed with #413

@conorsch please re-open if this still persist with the latest version
